### ISSUE TITLE
Potential error in kl loss computation for VAE tutorial

### ISSUE THREAD

**Issue by preethiseshadri518** [keras-team/keras-io#270]
Hi @fchollet,

I am following this VAE tutorial https://keras.io/examples/generative/vae/ and am confused regarding the KL loss computation. I am fairly new to VAEs so I realize that I may be wrong about this, but my understanding is that the KL loss term should be _summed_ instead of _averaged_ over the latent variables `z`. 

Currently the tutorial does the following:
```
kl_loss = 1 + z_log_var - tf.square(z_mean) - tf.exp(z_log_var)
kl_loss = tf.reduce_mean(kl_loss)
kl_loss *= -0.5
```
Based on what I described, I believe this should instead be:
```
kl_loss = 1 + z_log_var - tf.square(z_mean) - tf.exp(z_log_var)
kl_loss = tf.reduce_mean(tf.reduce_sum(kl_loss, axis=-1))
kl_loss *= 0.5
```
Many examples I've seen that cover the VAE loss derivation do not go into the implementation specifics, so perhaps averaging is completely fine. However, here are a few example posts I found that mention the loss being summed over the latent variables [1](https://stats.stackexchange.com/questions/394296/how-should-i-intuitively-understand-the-kl-divergence-loss-in-variational-autoen), [2](https://stats.stackexchange.com/questions/318748/deriving-the-kl-divergence-loss-for-vaes).

**Comment by fchollet** [comment]
What do you get if you make this change? In terms of results quality.

Fundamentally, this a flexible parameter: the KLD is a regularization term, and you can tweak its contribution. Here we have 2 variables so switching from mean to sum changes the contribution of the term by a factor 2. We have 2x less KLD regularization in the current example.

I'd be interested to see how the change affects results as you increase the rank of the latent space. With mean, the regularization term has a stable contribution independently of how many latent variables you have. With sum, you increase the regularization with the size of the latent space. Can you try it and report what you find?

**Comment by preethiseshadri518** [comment]
Makes sense! Let me run both and get back to you.

**Comment by sachinprasadhs** [comment]
@preethiseshadri518, Could you please update your findings. Thanks!
